report/report.go

```diff
@@ -52,3 +55,57 @@ func (rs *Set) GetMetadata() map[string]interface{} {
 type Handler interface {
 	HandleReport(io.Reader) error
 }
+
+// FilterReportSet by the logLevel
+func FilterReportSet(rs Set, logLevel string) Set {
+
+	elements := rs.GetElementArray()
+	filteredElements := make([]Element, 0, 0)
+	copy(filteredElements, elements)
+
+	for _, elm := range elements {
+		if logLevelIncludes(elm, logLevel) {
+			filteredElements = append(filteredElements, elm)
+		}
+	}
+
+	filteredSet := Set{Elements: filteredElements, Metadata: rs.Metadata}
+	return filteredSet
+}
+
+// LogLevelArray string denoting levels of logs
+var LogLevelArray = []string{"Debug", "Info", "Warn", "Error", "Fatal"}
+
+// logLevelIncludes utility function to know if one loglevel includes another
+func logLevelIncludes(elm Element, logLevel string) bool {
+
+	if elm.GetValue("Severity") == "" {
+		return false
+	}
+
+	levels := map[string]int{
+		"debug": 0,
+		"info":  1,
+		"warn":  2,
+		"error": 3,
+		"fatal": 4,
+	}
+
+	elmSeverity := strings.ToLower(elm.GetValue("Severity"))
```

Comment:
Perhaps do this at the top, and test elmSeverity != "", so as to reduce code duplication.